services/galley/src/Galley/API/MLS/Util.hs

```diff
@@ -50,10 +77,29 @@ getLocalConvForUser ::
   Local ConvId ->
   Sem r Data.Conversation
 getLocalConvForUser qusr lcnv = do
-  conv <- getConversation (tUnqualified lcnv) >>= noteS @'ConvNotFound
+  gtc <- getGlobalTeamConversationById lcnv
```

Comment:
I don't like the sideeffects introduced here into each lookup of a conversation. This looks badly designed.

Elsewhere, it seems global team conversation id is the same as the team Id (even though this is not documented anywhere; and it means there can only ever be one single global team conversation). Can't this mean you can tell earlier, if you have a user object available, you'd know their team, and already know which kind of conv you're dealing with, without an extra lookup?

And why does a GET of any conversation ID lead to an insert of a random user as creator of this conversation? That makes no sense semantically.

Reply:
We're taking this out in a follow-up  👍